pkg/receive/limiter.go

```diff
@@ -165,7 +173,9 @@ func (l *Limiter) loadConfig() error {
 		return false
 	}
 	if (l.receiverMode == RouterOnly || l.receiverMode == RouterIngestor) && seriesLimitIsActivated() {
-		l.HeadSeriesLimiter = NewHeadSeriesLimit(config.WriteLimits, l.registerer, l.logger)
+		l.headSeriesLimiterMtx.Lock()
+		l.headSeriesLimiter = NewHeadSeriesLimit(config.WriteLimits, l.registerer, l.logger)
```

Comment:
this simple synchronization seems like an ideal place to use an atomic value instead of a mutex